docs/architecture/adr/0006-white-label-architecture.md

```diff
@@ -0,0 +1,36 @@
+# White Label Architecture
+
+- Issue: [#7807](https://github.com/thunderbird/thunderbird-android/issues/7807)
+- Pull Request: [#7805](https://github.com/thunderbird/thunderbird-android/pull/7805)
+
+## Status
+
+- **Proposed**
+
+## Context
+
+Our project hosts two separate applications, K-9 Mail and Thunderbird for Android, which share a significant amount of functionality. Despite their common features, each app requires distinct branding elements such as app names, themes, and specific strings. To streamline our development process, a modular white-label approach is the desired approach. Each app is built as a module that relies on a shared codebase that should allows us to inject configuration details specific to each brand during the build or runtime. This will ensure that the core functionality remains consistent and maintainable.
```

Comment:
The last three sentences could probably be removed without losing any information. The information is repeated in the decision section (and that also feels like a better place for it).

```suggestion
Our project hosts two separate applications, K-9 Mail and Thunderbird for Android, which share a significant amount of functionality. Despite their common features, each app requires distinct branding elements such as app names, themes, and specific strings.
```